src/platform/pci.zig

```diff
@@ -145,8 +144,13 @@ fn function_scan(addr: Addr) void {
             os.log("Virtio display controller: Init has failed!\n", .{});
             return;
           };
+        if (os.drivers.vesa_log.get_info()) |vesa| {
           drv.modeset(os.drivers.vesa_log.framebuffer.?.bb_phys, vesa.width, vesa.height);
           os.drivers.vesa_log.set_updater(virtio_gpu.updater, @ptrToInt(drv));
+          os.log("Virtio display controller: Initialized with preexisting fb\n", .{});
+        } else {
+          os.drivers.vesa_log.register_fb(virtio_gpu.updater, @ptrToInt(drv), 800*4, 800, 600, 32);
+          drv.modeset(os.drivers.vesa_log.get_backbuffer_phy(), 800, 600);
```

Comment:
Have you tested this without a framebuffer tag? Pretty sure the `framebuffer.?` inside vesa_log will fail.

Reply:
The framebuffer is registered just before in `register_fb`

Reply:
Sure, what I'm saying is that we never call `register_fb()` if we don't have a framebuffer tag from the bootloader.